### ISSUE TITLE
Support easy collection of initial amounts/concentrations

### ISSUE THREAD

**Issue by exaexa** [LCSB-BioCore/SBML.jl#71]
....ideally with a nice API and auto-conversion between amounts and concentrations.

**Comment by exaexa** [comment]
cc @anandijain @paulflang
required by https://github.com/paulflang/SBMLToolkit.jl/pull/3

**Comment by paulflang** [comment]
This is how it is done atm: https://github.com/paulflang/SBML.jl/blob/1070601c451cdf6d8d5b4a5d0093e86ae4d341a0/src/reactionsystem.jl#L64
requires mutable SBML.Species, though. Any elegant suggestions how to do this without mutable structs?

```Julia
""" Convert initial_concentration to initial_amount """
function to_initial_amounts(model::Model)
    model = deepcopy(model)
    for specie in values(model.species)
        if isnothing(specie.initial_amount)
            compartment = model.compartments[specie.compartment]
            if !isnothing(compartment.size)
                specie.initial_amount = (specie.initial_concentration[1] * compartment.size, "")
            else
                @warn "Compartment $(compartment.name) has no `size`. Cannot calculate `initial_amount` of Species $(specie.name). Setting `initial_amount` to `initial_concentration`."
                specie.initial_amount = (specie.initial_concentration[1], "")
            end
            specie.initial_concentration = nothing
        end
    end
    model
end
```

**Comment by paulflang** [comment]
And this is how hOSU handling happens atm. It also needs work without mutable structs:
```Julia
""" Convert intensive to extensive mathematical expression """
function to_extensive_math!(model::SBML.Model)
    function conv(x::SBML.MathApply)
        SBML.MathApply(x.fn, SBML.Math[conv(x) for x in x.args])
    end
    function conv(x::SBML.MathIdent)
        x_new = x
        if x.id in keys(model.species)
            specie = model.species[x.id]
            if !specie.only_substance_units
                compartment = model.compartments[specie.compartment]
                if isnothing(compartment.size)
                    @warn "Specie $(x.id) hasOnlySubstanceUnits but its compartment $(compartment.name) has no size. Cannot auto-correct the rate laws $(x.id) is involved in. Please check manually."
                else
                    x_new = SBML.MathApply("/", SBML.Math[x,
                                SBML.MathVal(compartment.size)])
                    specie.only_substance_units = true
                end
            end
        end
        x_new
    end
    conv(x::SBML.MathVal) = x
    conv(x::SBML.MathLambda) =
        throw(DomainError(x, "can't translate lambdas to extensive units"))
    for reaction in values(model.reactions)
        reaction.kinetic_math = conv(reaction.kinetic_math)
    end
    model
end
```
These are the reasons that need to be fixed before we can get tests working. But there might be others, too. On that note, I'd like to point at #53 , without which we cannot tell if test are failing expectedly or due to bugs.

**Comment by exaexa** [comment]
This is now somehow fixed by #84 and #88

**Comment by paulflang** [comment]
well, the initial conditions part got fixed by #84 and #88. the `to_extensive_math!` part is still needed.
